sklearn/preprocessing/data.py

```diff
@@ -1979,13 +1984,27 @@ class QuantileTransformer(BaseEstimator, TransformerMixin):
 
     def __init__(self, n_quantiles=1000, output_distribution='uniform',
                  ignore_implicit_zeros=False, subsample=int(1e5),
-                 random_state=None):
+                 noise_variance=None, random_state=None):
         self.n_quantiles = n_quantiles
         self.output_distribution = output_distribution
         self.ignore_implicit_zeros = ignore_implicit_zeros
         self.subsample = subsample
+        self.noise_variance = noise_variance
         self.random_state = random_state
 
+    def _compute_quantile_one_column(self, X_col, references):
+        """Private function to compute the quantiles for one features."""
+        rng = check_random_state(self.random_state)
```

Comment:
Please pass the `rng` instance as an argument to `_compute_quantile_one_column` and instantiate it only once from `self.random_state` in the `fit` method.